open-craft/problem-builder
# Step Builder: Support for additional overlays for plot blocks

This PR follows up on #77 and adds functionality for defining custom overlays for plot blocks.

**Testing**

1. Add Step Builder to a unit.

2. Add two Mentoring Step blocks and a Review Step block to Step Builder.

3. Add two Rating questions to the first Mentoring Step block.

4. Add a Plot block to the second Mentoring Step block.

5. For each of the Rating questions, grab the corresponding "Question ID" from the EDIT dialog in Studio (last field, underneath "Label for high ratings").

6. Bring up the EDIT dialog for the Plot block and add the following to the "Claims and associated questions" field:

        This is an important claim, 7cf9575c53d5abc43fddedc4b06ae1a835dd4bf3, 79020e44630e246077a2bce92f45562fe24769cd

   Make sure to replace the two hashes with the question IDs you grabbed in the previous step.
7. Add one or more Plot Overlay blocks to the Plot block.

8. For each overlay, specify plot label and point color, and add a single pair `r1, r2` of coordinates to claim data, e.g.:

        3, 4

   Note that the numbers represent responses to the previously-added rating questions, so they must be between 1 and 5. (If you added multiple overlays in the previous step, you'll probably want to choose different numbers for their claim data so that corresponding points won't be stacked on top of each other when you enable the overlays for the plot in the LMS.)

9. Publish the unit.

10. Complete the first step of the unit by submitting answers to the rating questions.

11. Navigate to the second step. You should see a button on the right for each overlay you defined; the label of the button should correspond to the plot label you set for the overlay. The plot should not display data from overlays defined in previous steps.

12. Click the buttons to toggle the overlays. When you turn them on, the associated data should become visible on the plot. Point and border colors of individual overlays should match. Also, description and/or citation (if defined) should be shown on the right hand side for overlays that are currently visible. If overlays are off, the plot should not show associated points; description and citation should not be visible; border color of corresponding buttons should be reset to grey.
